nixos/modules/services/web-apps/stash.nix

```diff
@@ -0,0 +1,341 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (lib)
+    literalExpression
+    mkPackageOption
+    mkEnableOption
+    toUpper
+    mkOption
+    mkIf
+    head
+    optional
+    optionalString
+    ;
+
+  inherit (lib.types)
+    submodule
+    listOf
+    bool
+    attrs
+    port
+    package
+    str
+    ;
+  cfg = config.services.stash;
+
+  yamlFormat = pkgs.formats.yaml { };
+  settingsFile = yamlFormat.generate "config.yml" cfg.settings;
+
+  settingsType = submodule {
+    freeformType = yamlFormat.type;
+
+    options = {
+      jwt_secret_key = mkOption {
+        type = str;
+        description = ''
+          64 bytes long string.
+          Can be generated using {command}`openssl rand -hex 32`.'';
+      };
+      session_store_key = mkOption {
+        type = str;
+        description = ''
+          64 bytes long string.
+          Can be generated using {command}`openssl rand -hex 32`.'';
+      };
+      stash = mkOption {
+        type = listOf stashType;
+        description = ''
+          Add directories containing your adult videos and images. Stash will use these directories to find videos and images during scanning.
+        '';
+      };
+      ui = {
+        lastNoteSeen = mkOption {
+          type = str;
+          default = "20380119";
+          readOnly = true;
+          description = "Controls where to show changelog when site is opened";
+        };
+        frontPageContent = mkOption {
+          type = listOf attrs;
```

Comment:
Wanted to do `listOf ( either uiSavedFilterType uiPresetFilterType )` but can't, Since they have essentially the same type.

Related https://github.com/NixOS/nixpkgs/pull/254790